### PR TITLE
Prevent crashing with no content folder

### LINKED ISSUE #741: Missing content directory throws unhelpful error
### What version of `starlight` are you using?

0.10.1

### What version of `astro` are you using?

3.1.2

### What package manager are you using?

npm

### What operating system are you using?

Linux

### What browser are you using?

Firefox

### Describe the Bug

A missing content folder throws an obscure error. It's not an error I'd expect people to frequently encounter, but I this we can add a more useful error, just to cover this case.
```
11:48:31 [vite] Error when evaluating SSR module /node_modules/@astrojs/starlight/utils/routing.ts:
|- TypeError: (intermediate value).map is not a function
    at eval (/home/otterlord/Documents/Repos/api/test/node_modules/@astrojs/starlight/utils/routing.ts:10:66)
    at async instantiateModule (file:///home/otterlord/Documents/Repos/api/test/node_modules/vite/dist/node/chunks/dep-df561101.js:55974:9)

11:48:31 [vite] Error when evaluating SSR module /home/otterlord/Documents/Repos/api/test/node_modules/@astrojs/starlight/index.astro: failed to import "/node_modules/@astrojs/starlight/utils/routing.ts"
|- TypeError: (intermediate value).map is not a function
    at eval (/home/otterlord/Documents/Repos/api/test/node_modules/@astrojs/starlight/utils/routing.ts:10:66)
    at async instantiateModule (file:///home/otterlord/Documents/Repos/api/test/node_modules/vite/dist/node/chunks/dep-df561101.js:55974:9)

 error   (intermediate value).map is not a function
  File:
    /home/otterlord/Documents/Repos/api/test/node_modules/@astrojs/starlight/utils/routing.ts:38:66
  Code:
    37 | /** All entries in the docs content collection. */
    > 38 | const docs: StarlightDocsEntry[] = (await getCollection('docs')).map(({ slug, ...entry }) => ({
         |                                                                  ^
      39 |   ...entry,
      40 |   slug: normalizeIndexSlug(slug),
      41 | }));
  Stacktrace:
TypeError: (intermediate value).map is not a function
    at /home/otterlord/Documents/Repos/api/test/node_modules/@astrojs/starlight/utils/routing.ts:38:66
    at async instantiateModule (file:///home/otterlord/Documents/Repos/api/test/node_modules/vite/dist/node/chunks/dep-df561101.js:55974:9)
```

My proposal would be to check `getCollection` returns an array, then if not, throw an error about content not being found.

### Link to Minimal Reproducible Example

_No response_

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by TheOtterlord**
#### What kind of changes does this PR include?

- Changes to Starlight code

#### Description

- Closes #741 
- What does this PR change? Prevents Starlight crashing with an unhelpful error when `src/content/` doesn't exist. This would still show on the warning from `getCollection` (introduced in https://github.com/withastro/astro/pull/8640).

### CODE REVIEW

File: packages/starlight/utils/routing.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ interface Path extends GetStaticPathsItem {
 const normalizeIndexSlug = (slug: string) => (slug === 'index' ? '' : slug);
 
 /** All entries in the docs content collection. */
-const docs: StarlightDocsEntry[] = (await getCollection('docs')).map(({ slug, ...entry }) => ({
+const docs: StarlightDocsEntry[] = (await getCollection('docs'))?.map(({ slug, ...entry }) => ({
```
Comment (delucis) [inline packages/starlight/utils/routing.ts:38]: Thanks for reporting this and following up @TheOtterlord!

Does this really stop things crashing? I’d think that the optional chaining here would result in the `docs` variable being `undefined` and then we’d get similar problems downstream with places that use `docs`?

Comment (TheOtterlord) [reply packages/starlight/utils/routing.ts:38]: wait, how'd I mess a single line up :sweat_smile: 
this should be a nullcheck to make it an array. I'll update this now

**Comment by TheOtterlord** [comment]
Tested on docs with both `src/content/` deleted and with only `src/content/docs/` deleted

**Review by delucis — APPROVED** [review]
Ah yeah, this makes more sense. Thanks @TheOtterlord!

**Pull request merged**